EIPS/eip-7378.md

```diff
@@ -0,0 +1,79 @@
+---
+eip: 7378
+title: Add time-weighted averaging to the base fee
+description: Using geometric weights to average past block sizes into consideration
+author: Guy Goren (@guy-goren) <guy.nahsholim@gmail.com>
+discussions-to: https://ethereum-magicians.org/t/add-time-weighted-averaging-to-the-base-fee-mechanism/15142
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-07-22
+---
+
+## Abstract
+
+This EIP proposes a new formula for the base fee, derived from [EIP-1559](./eip-1559.md). The existing base fee formula,
+
+$$b[i+1]\triangleq  b[i] \cdot \left( 1+\frac{1}{8} \cdot \frac{s[i]-s^* }{s^* }\right)$$
+
+only considers the last block size $s[i]$. This mechanism incentivizes proposers to collude with users to manipulate the base fee.
+
+We propose to consider the history of block sizes via a weighted average with a geometric sequence as the weights. In particular, we suggest the following formula:
+
+$$b[i+1]\triangleq  b[i] \cdot \left( 1+\frac{1}{8} \cdot \frac{s_{\textit{avg}}[i]-s^* }{s^* }\right)$$
+
+where $s_{\textit{avg}}[i]$ is defined by:
+
+$$s_{\textit{avg}}[i] \triangleq \frac{1-q}{q}\sum_{k=1}^{\infty} q^k\cdot s[i-k+1], \hspace{1cm} q\in (0,1)$$ 
+
+## Motivation
+
+Mainly, to reduce bribe motivation (in particular in case the demand for blockspace is high (see Incentive Considerations section). Additionaly, to reduce oscillations and have a more stable fee setting mechanism.
+
+For determining which messages enter a block, a miner uses a mechnism that was proposed in EIP-1559. This mechanism introduced a base fee, which is a portion of the transaction fee that is burned and not awarded to miners, and which varies according to the fill rate of blocks. A target block size is defined, such that if a block is fuller than the target size, the base fee increases and if it is emptier, the base fee reduces.
```

Comment:
```suggestion
Proposers use a mechanism described in EIP-1559 to determine which messages to include in a block. This mechanism includes a "base fee": a portion of the transaction fee that is burned that varies according to the fill rate of blocks. A target block size is defined. If a block significantly exceeds the target size, the base fee increases, and if it is significantly smaller, the base fee lowers.
```
How about this phrasing? There are two non-grammatical errors in your original paragraph (miner -> proposer, mechnism -> mechanism).